### ISSUE TITLE
Integration test files naming requirement breaks VSCode test sidebar

### ISSUE THREAD

**Issue by vincent-hoodoo** [leancodepl/patrol#1995]
### Steps to reproduce

1. Create a patrol integration test
2. Open VSCode test sidebar
3. Click the "Run Tests" button

### Actual results

When using the VSCode test sidebar, the "Dart Code" plugin will search for all occurrences of files ending with `_test.dart`, and will present them in a list, making it easy to see at a glance which are passing and which are failing, and overall improving the DX when iterating over tests.

The dart extension assumes that all files ending with `_test.dart` can be run with the standard dart and Flutter test runners, so when it encounters a Patrol integration test, it tries to run it but just hangs indefinitely.

This makes it impossible to use the very useful "Run Tests" button in that panel:

![vscode_test_panel_run_tests_button](https://github.com/leancodepl/patrol/assets/104820536/2e9c4948-0219-45fe-905c-24beb84f526d)

Could the [requirement for the tests to end with `_test.dart`](https://github.com/leancodepl/patrol/blob/9f5cb3dd940c3905d7d042cb26daee283d1bb717/packages/patrol_cli/lib/src/test_finder.dart#L38) be removed? That way, teams using VSCode could freely rename their patrol test to another suffix, allowing them to use VSCode test panel

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### Patrol version

patrol: ^2.3.1

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
Program adb found in /Users/vincenthoodoo/Library/Android/sdk/platform-tools/adb
Env var $ANDROID_HOME set to /Users/vincenthoodoo/Library/Android/sdk
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /usr/local/bin/ideviceinstaller
```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
[!] Flutter (Channel [user-branch], 3.13.9, on macOS 14.1.2 23B92 darwin-x64, locale en-NZ)
    ! Flutter version 3.13.9 on channel [user-branch] at /Users/vincenthoodoo/Dev/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
    ! Upstream repository unknown source is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to unknown source to dismiss this error.
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.84.2)
[✓] Connected device (3 available)
[✓] Network resources
```

</details>

**Comment by bartekpacia** [comment]
Thanks for bringing this issue to our attention @vincent-hoodoo! We'll release a fix soon, likely as `patrol_cli v2.4.1`.


In the meantime I recommend you to upgrade to latest Flutter v3.16.x, and to latest Patrol v3.2.x.

**Comment by vincent-hoodoo** [comment]
Thank you @piotruela and @bartekpacia , I just implemented `test_file_suffix` in my patrol configuration and it's working well, I can now run my test suite from VSCode.
I noticed that https://github.com/leancodepl/patrol/pull/2009 didn't include any documentation update, was there a reason for it? I'm probably not the only one with this issue, someone else might be wondering how to address it.
